be/src/column/map_column.cpp

```diff
@@ -425,23 +425,27 @@ bool MapColumn::equals(size_t left, const starrocks::Column& rhs, size_t right)
     for (uint32_t i = lhs_offset; i < lhs_end; ++i) {
         bool found = false;
         for (uint32_t j = rhs_offset; j < rhs_end; ++j) {
-            if (!_keys->equals(i, *(rhs_map._keys.get()), j)) {
+            int res = _keys->equals(i, *(rhs_map._keys.get()), j, safe_eq);
+            if (res == -1) {
+                return -1;
```

Comment:
should not directly return
{null:1, 'a':2} == {null:2,'a':2} => false
{null:1, 'a':2} == {null:1, 'a':3} => false
{null:1, 'a':2} == {null:1, 'b':2} => false